### PR TITLE
Ensure required options are only enforced for relevant subcommands in SDK

### PR DESCRIPTION
**Pull request by shackra**
## Summary by CodeRabbit

- **New Features**
	- Introduced new commands for the CLI: `grpcServerCmd` to start the connector service and `capabilitiesCmd` to retrieve connector capabilities.
	- Added persistence options to various fields in the CLI configuration.

- **Bug Fixes**
	- Enhanced validation logic for command execution, ensuring required arguments are enforced correctly.

- **Documentation**
	- Improved clarity and maintainability of command definitions and flag settings within the CLI.

- **Tests**
	- Added unit tests for validating sub-command behaviors in the CLI.

### CODE REVIEW

**Comment by ggreer** [comment]
I tried this branch with baton-ldap and it doesn't seem to get config options from env vars. Also if a required field is missing, it errors but doesn't exit. I had to ctrl+c to stop the process.

```
$ BATON_PASSWORD=admin BATON_BASE_DN='dc=example,dc=org' BATON_USER_DN='CN=admin,DC=example,DC=org' BATON_DOMAIN='localhost' ./dist/darwin_arm64/baton-ldap --log-level=debug
{"level":"debug","ts":1726846371.912221,"caller":"logging/logging.go:62","msg":"Logger created!","log_level":"debug"}
{"level":"debug","ts":1726846371.9126122,"caller":"connector/connector.go:81","msg":"creating new LDAP connector","serverUrl":"ldap://localhost","baseDN":"dc=example,dc=org","disableOperationalAttrs":false}
{"level":"debug","ts":1726846371.914392,"caller":"ldap/client.go:250","msg":"Binding to LDAP server authenticated"}
{"level":"debug","ts":1726846371.918334,"caller":"connectorrunner/runner.go:139","msg":"runner: claiming worker"}
{"level":"debug","ts":1726846371.918352,"caller":"connectorrunner/runner.go:147","msg":"runner: worker claimed, checking for next task"}
{"level":"debug","ts":1726846371.918367,"caller":"connectorrunner/runner.go:174","msg":"runner: got task","task_id":"","task_type":"sync_full"}
{"level":"debug","ts":1726846371.91837,"caller":"connectorrunner/runner.go:178","msg":"runner: one-shot mode enabled. Performing action synchronously."}
{"level":"debug","ts":1726846371.9209049,"caller":"connector/connector_server_unix.go:39","msg":"listener started","listen_port":60103,"listener_fd":10}
at least one of the flags in the group [domain url] is required
{"level":"error","ts":1726846371.935843,"caller":"connector/connector.go:254","msg":"connector service quit unexpectedly","error":"exit status 1"}


^C{"level":"error","ts":1726846382.057437,"caller":"connector/connector.go:257","msg":"error closing connector wrapper","error":"error closing connector service stdin: close |1: file already closed"}
{"level":"error","ts":1726846382.057388,"caller":"connectorrunner/runner.go:182","msg":"runner: error processing on-demand task","error":"runner: error creating connector client: context canceled","task_id":"","task_type":"sync_full"}
{"level":"error","ts":1726846382.057545,"caller":"cli/commands.go:164","msg":"error running connector","error":"runner: error creating connector client: context canceled"}
runner: error creating connector client: context canceled

```

**Comment by shackra** [comment]
@ggreer I added code that visits the flags of the other two subcommands and sets their values if Viper know anything about the flag, just like we do for the main subcommand. It should fix what you just saw with baton LDAP

File: internal/tests/notrequireargsinsubcommands_test.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+			requiredField,
+		},
+	)
+
+	t.Run("should run «help» sub-command successfully", func(t *testing.T) {
+		_, err := entrypoint(ctx, carrier, "help")
+		require.NoError(t, err)
+	})
+
+	t.Run("should run «capabilities» sub-command without success", func(t *testing.T) {
```
Comment (mgaeta) [inline internal/tests/notrequireargsinsubcommands_test.go:27]: Can you add a test where you call `entrypoint()` with `"capabilities"` and then _not_ error?
Something like:
```go
t.Run("should run «capabilities» sub-command without success", func(t *testing.T) {
  _, err := entrypoint(ctx, carrier, "capabilities", "name")
  require.NoError(t, err)
})
```

Comment (shackra) [reply internal/tests/notrequireargsinsubcommands_test.go:27]: I can, and I should!

File: pkg/field/defaults.go [hunk 2]
```diff
@@ -3,32 +3,32 @@ package field
...
-	rotateCredentialsTypeField = StringField("rotate-credentials-type", WithHidden(true), WithDescription("The type of the resource to rotate credentials on"))
-	ticketIDField              = StringField("ticket-id", WithHidden(true), WithDescription("The ID of the ticket to get"))
-	ticketTemplatePathField    = StringField("ticket-template-path", WithHidden(true), WithDescription("A JSON file describing the ticket to create"))
-	logLevelField              = StringField("log-level", WithDefaultValue("info"), WithDescription("The log level: debug, info, warn, error"))
-	skipFullSync               = BoolField("skip-full-sync", WithDescription("This must be set to skip a full sync"))
+		"and write access is required. Defaults to the OS temporary directory."), WithPersistent(true))
+	clientIDField              = StringField("client-id", WithDescription("The client ID used to authenticate with ConductorOne"), WithPersistent(true))
+	clientSecretField          = StringField("client-secret", WithDescription("The client secret used to authenticate with ConductorOne"), WithPersistent(true))
```
[the remark this answers is not in the document]
Comment (shackra) [reply pkg/field/defaults.go:14]: I'm 100% this is not about storing data on disk (or anywhere for that matter), @coderabbitai

**Review by ggreer — APPROVED** [review]

**Pull request merged**